Validation/RecoTrack/plugins/MultiTrackValidator.cc

```diff
@@ -459,6 +461,44 @@ size_t MultiTrackValidator::tpDR(const TrackingParticleRefVector& tPCeff,
   return n_selTP_dr;
 }
 
+
+size_t MultiTrackValidator::tpDR_jet(const TrackingParticleRefVector& tPCeff,
+                                 const std::vector<size_t>& selected_tPCeff,
+                                 DynArray<float>& dR_tPCeff_jet,
+                                 const edm::View<reco::Candidate>* const & cores) const {
+  float etaL[tPCeff.size()], phiL[tPCeff.size()];
+  size_t n_selTP_dr = 0;
+  for(size_t iTP: selected_tPCeff) {
+    //calculare dR wrt inclusive collection (also with PU, low pT, displaced)
+    auto const& tp2 = *(tPCeff[iTP]);
+    auto  && p = tp2.momentum();
+    etaL[iTP] = etaFromXYZ(p.x(),p.y(),p.z());
+    phiL[iTP] = atan2f(p.y(),p.x());
+  }
+  for(size_t iTP1: selected_tPCeff) {
+    auto const& tp = *(tPCeff[iTP1]);
+    double dR = std::numeric_limits<double>::max();
+    if(dRtpSelector(tp)) {//only for those needed for efficiency!
+      ++n_selTP_dr;
+      float eta = etaL[iTP1];
+      float phi = phiL[iTP1];
+      for (unsigned int ji = 0; ji < cores->size(); ji++) {//jet loop
+        if((*cores)[ji].pt() > ptMinJet_){
```

Comment:
I'd prefer that this cut would be made with a (generic ref) selector outside MTV.

Reply:
@makortel I guess in the log term we could also think of binning rather than cutting on Jet pT, so I'd rather keep the cut inside. What would be the advantage of an additional external selection?

Reply:
@arizzi I was mainly thinking to avoid extending the MTV complexity when something could be done outside of it. Also currently all other cuts than the "standard" TrackingParticle efficiency denominator cuts are done outside MTV. A generic jet selector outside MTV would also provide an easy placeholder for any other cuts on jets (eta, phi, who knows). Binning cuts are certainly necessary to do here, I agree. 

At minimum the minimal cuts on jets (now minimum pT) should be done only once, i.e. moved outside of the loop over `selected_tPCeff`.